qa/rpc-tests/zkey_import_export.py

```diff
@@ -0,0 +1,185 @@
+
+            def cmp_confirmations_high_to_low(a, b):
+                return cmp(b["amount"], a["amount"])
+
+            txs.sort(cmp_confirmations_high_to_low)
+            print("Sorted txs", txs)
+            print("amts", amts)
+
+            try:
+                assert_equal(amts, [tx["amount"] for tx in txs])
+            except AssertionError:
+                logging.error(
+                    'Expected amounts: %r; txs: %r',
+                    amts, txs)
+                raise
+
+        def get_private_balance(node):
+            balance = node.z_gettotalbalance()
+            return balance['private']
+
+        def find_imported_key(node, import_zaddr):
+            zaddrs = node.z_listaddresses()
+            for zdr in zaddrs:
+                if zdr == import_zaddr:
+                    return zdr
+                else:
+                    print("Imported zaddr {0} not found".format(import_zaddr))
+
+        # Seed Alice with some funds
+        alice.generate(10)
+        self.sync_all()
+        miner.generate(100)
+        self.sync_all()
+        # Shield Alice's coinbase funds to her zaddr
+        alice_zaddr = alice.z_getnewaddress()
+        res = alice.z_shieldcoinbase("*", alice_zaddr)
+        txid = self.wait_and_assert_operationid_status(alice, res['opid'])
+        miner.generate(6)
+        self.sync_all()
+        # List funds
+        funds = alice.z_listreceivedbyaddress(alice_zaddr)
+        # print("Alice's funds after shield", funds)
+
+        # Ensure new funds have been confirmed
+        miner.generate(2)
+        self.sync_all()
+
+        # Now get a pristine z-address for receiving transfers:
+        bob_zaddr = bob.z_getnewaddress()
+        print("bob zaddr set:", bob_zaddr)
+        verify_utxos(bob, [], bob_zaddr)
+        # TODO: Verify that charlie doesn't have funds in addr
+        # verify_utxos(charlie, [])
+
+        # the amounts of each txn embodied which generates a single UTXO:
+        amounts = map(Decimal, ['2.3', '3.7', '0.1', '0.5', '1.0', '0.19'])
+
+        # Internal test consistency assertion:
+        assert_greater_than(
+            get_private_balance(alice),
+            reduce(Decimal.__add__, amounts))
+
+        logging.info("Sending pre-export txns...")
+        for amount in amounts[0:2]:
+            alice_to_bob(amount)
+
+        logging.info("Exporting privkey from bob...")
+        privkey = bob.z_exportkey(bob_zaddr)
+
+        logging.info("Sending post-export txns...")
+        for amount in amounts[2:4]:
+            alice_to_bob(amount)
+
+        print("Bob amounts:", amounts[:4])
+        verify_utxos(bob, amounts[:4], bob_zaddr)
+        # verify_utxos(charlie, [])
+
+        logging.info("Importing privkey into charlie...")
+        # z_importkey rescan defaults to "whenkeyisnew", so should rescan here
+        charlie.z_importkey(privkey)
+        ipk_zaddr = find_imported_key(charlie, bob_zaddr)
+        assert_equal(bob_zaddr, ipk_zaddr)
```

Comment:
This assertion would be redundant if `find_imported_key` asserted when it failed to find the address.